### ISSUE TITLE
Add support for the new optional fields API

### ISSUE THREAD

**Issue by vmarcosp** [green-labs/ppx_ts#16]
Hey team! 
First off, thanks for your awesome work and contributions to the rescript ecosystem <3 

I'm using this ppx to build the new version of [Ancestor](https://github.com/rescriptbr/ancestor) and one ppx that could save me a lot of work is _partial_. However, partial doesn't support the new optional fields api introduced on the rescript @ v10. 
Are you planning to add support for the new optional fields api? If so, can I help you with this feature?

**Comment by mununki** [comment]
@vmarcosp Hey!
Thank you for using this ppx.
Sure, it would support the optional field record. I couldn't grab the time to fix it yet. PR is always welcome! 👍

**Comment by vmarcosp** [comment]
Nice!
Sure, I can help with this issue :)
Are you planning to keep supporting both apis using ?: and option<a> or should we create a second partial implementation?

**Comment by mununki** [comment]
Good question!
I think the partial ppx should support the `?` not `option<'a>` which is now possible with the new feature of v10. I prefer not to make a breaking change, but this time breaking would be okay.
What do you think?

**Comment by vmarcosp** [comment]
I agree, it's ok to me make a breaking since ppl can still using the current version instead of migrating to a new one.
